### PR TITLE
Use provided proxy to download chrome binary

### PR DESCRIPTION
**Pull request by Mzack9999**
## Description
This PR uses the provided proxy from chrome in the launcher to configure the http client at `lib/launcher/browser.go`, allowing chrome binaries to be downloaded through a custom proxy. Previously, the custom http client created with the function `httpClient()` in `browser.go` performed direct connections as the `Transport.Proxy` was nil.

### CODE REVIEW

File: lib/launcher/browser.go [hunk 1]
```diff
@@ -229,7 +244,16 @@ func (lc *Browser) download(ctx context.Context, u string) error {
 }
 
 func (lc *Browser) httpClient() *http.Client {
-	return &http.Client{Transport: &http.Transport{DisableKeepAlives: true}}
+	transport := &http.Transport{
+		DisableKeepAlives: true,
+		TLSClientConfig: &tls.Config{
+			InsecureSkipVerify: true,
```
Comment (ysmood) [inline lib/launcher/browser.go:250]: Why do you want to disable it?

**Comment by ysmood** [comment]
Please fix the CI errors.

File: lib/launcher/browser.go (hunk 1 above)
Comment (Mzack9999) [reply lib/launcher/browser.go:250]: Many proxies (for example https://portswigger.net/burp) provide their own self-signed certificate. Would you like me to make this optional?

Comment (ysmood) [reply lib/launcher/browser.go:250]: Better to make it opotional

**Comment by Mzack9999** [comment]
> Please fix the CI errors and add unit tests for it. Run the `go run ./lib/utils/simple-check` before you push.

Added unit tests, unfortunately I'm a bit out of ideas how to reach 100% of coverage xD

**Comment by ysmood** [comment]
Please try to fix the CI error

**Comment by ysmood** [comment]
You might need to add a test cert for test the branch.

**Review by ysmood — APPROVED** [review]

**Pull request merged**